show-device-location-using-indoor-positioning/src/main/java/com/esri/arcgismaps/sample/showdevicelocationusingindoorpositioning/MainActivity.kt

```diff
@@ -0,0 +1,365 @@
+        if (featureTable.tableName == "IPS_Positioning") {
+            val positioningFeatureTable = featureTable as ServiceFeatureTable
+            // when multiple entries are available, IndoorsLocationDataSource constructor function
+            // looks up the entry with the most recent date and takes this positioning data
+            // set up queryParameters to grab one result.
+            val dateCreatedFieldName =
+                getDateCreatedFieldName(positioningFeatureTable.fields)
+            if (dateCreatedFieldName == null) {
+                showError("The service table does not contain \"DateCreated\" fields.")
+                return
+            }
+            val queryParameters = QueryParameters().apply {
+                // set a limit of 1 on the number of returned features per request
+                maxFeatures = 1
+                // 1=1 is a true where clause which will result in all matching records being returned
+                whereClause = "1 = 1"
+                // find and sort out the orderByFields by most recent first
+                orderByFields.add(
+                    OrderBy(
+                        dateCreatedFieldName,
+                        sortOrder = SortOrder.Descending
+                    )
+                )
+            }
+            lifecycleScope.launch {
+                positioningFeatureTable.queryFeatures(queryParameters)
+                    .onSuccess { queryResults ->
+                        // perform search query using the queryParameters
+                        queryResults.forEach { feature ->
+                            // check if serviceFeatureTable contains positioning data
+                            // The ID that identifies a row in the positioning table.
+                            val globalID =
+                                feature.attributes[positioningFeatureTable.globalIdField].toString()
+                            val positioningId = Guid(globalID)
+                            // Setting up IndoorsLocationDataSource with positioning, pathways tables and positioning ID.
+                            // positioningTable - the "ips_positioning" feature table from an IPS-enabled map.
+                            // pathwaysTable - An ArcGISFeatureTable that contains pathways as per the ArcGIS Indoors Information Model.
+                            // Setting this property enables path snapping of locations provided by the IndoorsLocationDataSource.
+                            // levelsTable - An ArcGISFeatureTable that contains floor levels in accordance with the ArcGIS Indoors Information Model.
+                            // Providing this table enables the retrieval of a location's floor level ID.
+                            // positioningID - an ID which identifies a specific row in the positioningTable that should be used for setting up IPS.
+                            mIndoorsLocationDataSource = IndoorsLocationDataSource(
```

Comment:
Yes, that seems to be the case since the `maxFeatures` is set to 1, so the `mIndoorsLocationDataSource` will be set once. But the code doesn't have to be looping through either, so could be updated with: 

```kotlin
.onSuccess { queryResults ->
                        val featureResult = queryResults.first()
```